poetry/installation/authenticator.py

```diff
@@ -0,0 +1,133 @@
+from typing import TYPE_CHECKING
+
+from poetry.utils._compat import urlparse
+from poetry.utils.password_manager import PasswordManager
+
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Optional
+    from typing import Tuple
+
+    from clikit.api.io import IO
+    from requests import Request  # noqa
+    from requests import Response  # noqa
+    from requests import Session  # noqa
+
+    from poetry.config.config import Config
+
+
+class Authenticator(object):
+    def __init__(self, config, io):  # type: (Config, IO) -> None
+        self._config = config
+        self._io = io
+        self._session = None
+        self._credentials = {}
+        self._password_manager = PasswordManager(self._config)
+
+    @property
+    def session(self):  # type: () -> Session
+        from requests import Session  # noqa
```

Comment:
Would be great to avoid these if possible - both lazy initialisatio and deferred imports. Unless this provides a significant improvement.

Reply:
Actually, I plan on making more imports lazy.

The reasoning behind this is that imports are costly for CLI tools and we should defer them as much as possible.